shell/src/main/java/alluxio/cli/fsadmin/FileSystemAdminShellUtils.java

```diff
@@ -37,6 +47,29 @@ public static int compareTierNames(String a, String b) {
     return aValue - bValue;
   }
 
+  /**
+   * Checks if the master client service is running.
+   * Throws an exception if fails to determine that the master client service is running.
+   */
+  public static void checkMasterClientServiceIsRunning() throws IOException {
+    try (CloseableResource<FileSystemMasterClient> client =
+        FileSystemContext.get().acquireMasterClientResource()) {
+      try {
+        InetSocketAddress address = client.get().getAddress();
+        System.out.println(address.getHostName());
+
+        List<InetSocketAddress> addresses = Arrays.asList(address);
+        MasterInquireClient inquireClient = new PollingMasterInquireClient(addresses, () ->
+            new ExponentialBackoffRetry(50, 100, 2)
+        );
+        inquireClient.getPrimaryRpcAddress();
+      } catch (UnavailableException e) {
+        throw new IOException("The Alluxio leader master is not currently serving requests, "
```

Comment:
what if the leader master is serving requests, but this client doesn't have access to it due to a firewall, or being configured with the wrong address?

Reply:
@aaudiber We change the exception to : Cannot connect to Alluxio leader master?
This can include all the cases: leader master is not started/ not serving requests/ don't have acess

Reply:
sgtm